### PR TITLE
Second delivery

### PR DESCRIPTION
**Pull request by yoelfme**

### CODE REVIEW

File: src/App.style.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import styled from 'styled-components';
+
+export const Header = styled.header`
+  ul {
```
Comment (timgivois) [inline src/App.style.js:4]: don't create one component with global css

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
+      <Provider store={store}>
+        <BrowserRouter>
+          <div className="App">
```
Comment (timgivois) [inline src/App.js:15]: delete classNames

**Comment by timgivois** [comment]
don't create different PRs, and different branches

**Comment by timgivois** [comment]
you have a lot of linting problems

**Comment by yoelfme** [comment]
Thanks for your feedback, I'm going to fix some problem and merge this branch with the first branch that I created

File: src/middleware/saveFavorites.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import gifsActions from '../actions/gifs';
+
+const { types } = gifsActions;
+
+const saveFavorites = store => next => action => {
```
Comment (Robfz) [inline src/middleware/saveFavorites.js:5]: 👍

File: src/reducers/gifs.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      } = payload;
+
+      const mappedGifs = gifs.reduce((acc, gif) => {
+        acc[gif.id] = gif;
+        return acc;
+      }, {});
+
+      return { ...state, gifs: mappedGifs, loading: true };
+    }
+    case types.LOAD_DATA_FAILED: {
```
Comment (Robfz) [inline src/reducers/gifs.js:35]: I think this reducer is too big, it handles different business logic. Always try to make them as atomic as possible 👍

File: src/sagas/index.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  loadDataFinished,
+  loadDataFailed,
+  searchFinished,
+  searchFailed,
+  dontSearch,
+  loadFavoritesFinished
+} = gifsActions.creators;
+
+const API_URL = 'https://api.giphy.com/v1/gifs/';
+const API_KEY = 'PCv31jAmmHGTcGvY06Tow2lcIHZPMmj6';
```
Comment (Robfz) [inline src/sagas/index.js:17]: As a general tip, never commit api keys or secrets into version control. Get them from environmental variables.

File: src/sagas/index.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    const gifs = yield getTrendingGifs();
+
+    yield put(loadDataFinished({ gifs }));
+  } catch (e) {
+    yield put(loadDataFailed({ error: e }));
+  }
+}
+
+function* loadFavorites() {
+  const favorites = JSON.parse(localStorage.getItem('favorites')) || {};
```
Comment (Robfz) [inline src/sagas/index.js:40]: 👍

File: src/sagas/index.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  try {
+    const gifs = yield getTrendingGifs();
+
+    yield put(loadDataFinished({ gifs }));
+  } catch (e) {
+    yield put(loadDataFailed({ error: e }));
+  }
+}
+
+function* loadFavorites() {
```
Comment (Robfz) [inline src/sagas/index.js:39]: You could also implement the `saveFavorites` as a saga.

**Pull request closed without merging**